Backbone.Tests.ArchUnit/DomainEvents.cs

```diff
@@ -0,0 +1,57 @@
+﻿using ArchUnitNET.Domain.Extensions;
+using ArchUnitNET.Fluent.Conditions;
+using ArchUnitNET.xUnit;
+using Backbone.BuildingBlocks.Domain.Events;
+using Backbone.Tooling.Extensions;
+using static ArchUnitNET.Fluent.ArchRuleDefinition;
+
+namespace Backbone.Backbone.Tests.ArchUnit;
+
+public class DomainEvents
+{
+    [Fact]
+    public void IncomingDomainEventsShouldNotHaveConstructors()
+    {
+        Types()
+            .That().AreAssignableTo(typeof(DomainEvent))
+            .And().ResideInNamespace("Backbone.Modules.*.Domain.DomainEvents.Incoming.*", true)
+            .Should().FollowCustomCondition(type =>
+            {
+                if (type.GetConstructors().Any(c => c.Parameters.Any()))
+                    return new ConditionResult(type, false, "should not have constructors");
+
+                return new ConditionResult(type, true);
+            }, "")
+            .Because("otherwise the JSON Serializer would call the constructor instead of serializing directly, which would cause an invalid serialization")
```

Comment:
```suggestion
            .Because("objects are only created by the JSON Serializer anyway. Adding a constructor with parameters only causes errors during deserialization if the parameter names don't match the property names.")
```